### PR TITLE
reusable polling retry executor & add td store_last_result download retry

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

**Comment by danielnorberg** [comment]
@komamitsu PTAL

File: digdag-standards/src/main/java/io/digdag/standards/operator/PollingRetryExecutor.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            result = f.call();
+        }
+        catch (Exception e) {
+            if (!retryPredicate.test(e)) {
+                throw new TaskExecutionException(e, TaskExecutionException.buildExceptionErrorConfig(e));
+            }
+
+            int retry = retryState.get(RETRY, int.class, 0);
+            retryState.set(RETRY, retry + 1);
+            int interval = (int) Math.min(minRetryInterval.getSeconds() * Math.pow(2, retry), maxRetryInterval.getSeconds());
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/PollingRetryExecutor.java]: How about extracting this calculation as a method of `DurationInterval`?

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java [hunk 2]
```diff
@@ -410,4 +414,27 @@ static JobState empty()
             return ImmutableJobState.builder().build();
         }
     }
+
+    static final Duration DEFAULT_MIN_POLL_INTERVAL = Duration.ofSeconds(1);
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java]: You can put `private` keyword here

File: digdag-standards/src/main/java/io/digdag/standards/operator/td/TdDdlOperatorFactory.java [hunk 3]
```diff
@@ -109,25 +107,15 @@ public TaskResult runTask(TaskExecutionContext ctx)
...
-                        int interval = (int) Math.min(INITIAL_RETRY_INTERVAL * Math.pow(2, retry), MAX_RETRY_INTERVAL);
-                        state.set("retry", retry + 1);
-                        state.set("operation", i);
-                        throw TaskExecutionException.ofNextPolling(interval, ConfigElement.copyOf(state));
-                    }
-                    state.remove("retry");
+                    pollingRetryExecutor(state, "retry")
+                            .retryUnless(TDOperator::isDeterministicClientException)
+                            .withRetryInterval(retryInterval)
+                            .withErrorMessage("DDL operation failed")
```
Comment (komamitsu) [inline digdag-standards/src/main/java/io/digdag/standards/operator/td/TdDdlOperatorFactory.java]: It looks better to add the operation into the message like
```
.withErrorMessage("DDL operation failed [{}]", operation)
```

**Pull request closed without merging**